core/src/com/mygdx/game/steerers/FollowPathSteerer.java

```diff
@@ -136,21 +206,61 @@ public void stopSteering () {
 	public void finishSteering () {
 		pathToRender.clear();
 	}
-	
+
+	boolean deadlockDetection;
+	float deadlockDetectionStartTime;
+	float collisionDuration;
+	static final float deadlockTime = .5f;
+	static final float maxNoCollisionTime = deadlockTime + .5f;
+
 	@Override
 	public void onSteering () {
+		
+		if (prioritySteering.getSelectedBehaviorIndex() == 0) {
+				float pr = proximity.getRadius() * 1.5f;
+				if (linePath.getEndPoint().dst2(steerableBody.getPosition()) <= pr * pr) {
+					// Disable collision avoidance near the end of the path since the obstacle 
+					// will likely prevent the entity from reaching the target.
+					collisionAvoidanceSB.setEnabled(false);
+					deadlockDetectionStartTime = Float.POSITIVE_INFINITY;
+				}
+				else if (deadlockDetection) {
+					// Accumulate collision time during deadlock detection
+					collisionDuration += GdxAI.getTimepiece().getDeltaTime();
+
+					if (GdxAI.getTimepiece().getTime() - deadlockDetectionStartTime > deadlockTime && collisionDuration > deadlockTime *.6f) {
+						// Disable collision avoidance since most of the deadlock detection period has been spent on collision avoidance
+						collisionAvoidanceSB.setEnabled(false);
+					}
+				}
+				else {
+					// Start deadlock detection
+					deadlockDetectionStartTime = GdxAI.getTimepiece().getTime();
+					collisionDuration = 0;
+					deadlockDetection = true;
+				}
+		}
+		else {
+			if (deadlockDetection && !collisionAvoidanceSB.isEnabled() && GdxAI.getTimepiece().getTime() - deadlockDetectionStartTime > maxNoCollisionTime) {
+				collisionAvoidanceSB.setEnabled(true);
+				deadlockDetection = false;
+			}
+		}
+
 		// Check if steering target path segment changed.
 		int traversedSegment = followPathSB.getPathParam().getSegmentIndex();
 		if (traversedSegment > currentSegmentIndex) {
 			// Update model target orientation. Current orientation wi
 			currentSegmentIndex = traversedSegment;
-			Segment<Vector3> segment = linePath.getSegments().get(currentSegmentIndex);
-			steerableBody.setModelTargetOrientation(segment.getEnd().x - segment.getBegin().x, segment.getEnd().z - segment.getBegin().z);
+/*
+//			Segment<Vector3> segment = linePath.getSegments().get(currentSegmentIndex);
+//			steerableBody.setModelTargetOrientation(segment.getEnd().x - segment.getBegin().x, segment.getEnd().z - segment.getBegin().z);
 			// Update current navmesh triangle
 			steerableBody.currentTriangle = navMeshPointPath.getToTriangle(currentSegmentIndex);
 			// Set model to be visible on the same layer as mesh part index of current triangle
 			steerableBody.visibleOnLayers.clear();
 			steerableBody.visibleOnLayers.set(steerableBody.currentTriangle.meshPartIndex);
+ */
```

Comment:
Commented code is now executed by the caller after invoking this method.
Also, model orientation is now based on the direction of motion rather than the direction of current segment. This makes collision avoidance more realistic.